.github/workflows/canary-integration-test.yml

```diff
@@ -1468,3 +1468,44 @@ jobs:
         uses: ./.github/workflows/collect-logs
         with:
           name: csi-hostnetwork-disabled
+
+  csi-addons-testing:
+      runs-on: ubuntu-20.04
+      if: "!contains(github.event.pull_request.label.*.name, 'skip-ci')"
+      steps:
+        - name: checkout
+          uses: actions/checkout@v4
+          with:
+            fetch-depth: 0
+
+        - name: consider debugging
+          uses: ./.github/workflows/tmate_debug
+          with:
+            use-tmate: ${{ secrets.USE_TMATE }}
+
+        - name: setup cluster resources
+          uses: ./.github/workflows/canary-test-config
+
+        - name: use local disk
+          run: tests/scripts/github-action-helper.sh use_local_disk
+
+        - name: create cluster prerequisites
+          run:  tests/scripts/github-action-helper.sh create_cluster_prerequisites
+
+        - name: deploy cluster
+          run: tests/scripts/github-action-helper.sh deploy_cluster
+
+        - name: setup csi-addons
+          run:  tests/scripts/csiaddons.sh setup_csiaddons
+
+        - name: wait for prepare pod
+          run: tests/scripts/github-action-helper.sh wait_for_prepare_pod
+
+        - name: wait for ceph to be ready
+          run: tests/scripts/github-action-helper.sh wait_for_ceph_to_be_ready all 1
+
+        - name: verify csiaddonsnodes crd created
+          run:  tests/scripts/csiaddons.sh verify_crd_created
+
+        - name: verify csi-addons container running
+          run: tests/scripts/csiaddons.sh verify_container_is_running
```

Comment:
I think there should a wait involved somewhere and verify container running before checking for CR.
sometimes it may take longer to pull the image.

Reply:
Yea, It did occur to me. So I moved the `setup csi-addons` step after `deploy-cluster`.
So we have `wait for prepare pod` and `wait for ceph to be ready` steps running which takes ~3min before verifying the CR. 
I think the node has enough time to pull the image.